extensions/granola/src/browse-folders.tsx

```diff
@@ -0,0 +1,234 @@
+        if (!response || !response.lists || typeof response.lists !== "object") {
+          console.error("Unexpected response format from getFolders()", response);
+          showFailureToast({
+            title: "Failed to load folders",
+            message: "Received an unexpected response format from Granola API",
+          });
+          setFolders([]);
+          setIsLoading(false);
+          return;
+        }
+
+        const foldersList = Object.values(response.lists);
+        setFolders(foldersList);
+      } catch (error) {
+        showFailureToast({ title: "Failed to load folders", message: String(error) });
+        setFolders([]);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    loadFolders();
+  }, []);
+
+  return (
+    <List isLoading={isLoading} searchBarPlaceholder="Search folders...">
+      {folders.length === 0 && !isLoading ? (
+        <List.EmptyView
+          icon={{ source: Icon.Folder, tintColor: Color.Blue }}
+          title="No Folders Found"
+          description="You don't have any folders yet. Create folders in Granola to organize your notes."
+          actions={
+            <ActionPanel>
+              <Action.OpenInBrowser title="Open Granola App" url="granola://" icon={Icon.AppWindow} />
+            </ActionPanel>
+          }
+        />
+      ) : (
+        folders.map((folder) => (
+          <List.Item
+            key={folder.id}
+            title={folder.title}
+            subtitle={
+              folder.description
+                ? folder.description.trim().length > 80
+                  ? folder.description.trim().slice(0, 77) + "..."
+                  : folder.description.trim()
+                : ""
+            }
+            icon={{
+              source: folder.icon ? mapIconToRaycast(folder.icon.value) : Icon.Folder,
+              tintColor: folder.icon ? mapColorToRaycast(folder.icon.color) : Color.Blue,
+            }}
+            accessories={[{ text: `${folder.document_ids.length} notes` }]}
+            actions={
+              <ActionPanel>
+                <Action.Push
+                  title="View Notes in Folder"
+                  icon={Icon.Document}
+                  target={<DocumentsList folderTitle={folder.title} documentIds={folder.document_ids} />}
+                />
+              </ActionPanel>
+            }
+          />
+        ))
+      )}
+    </List>
+  );
+};
+
+// Display documents in a folder
+function DocumentsList({ folderTitle, documentIds }: { folderTitle: string; documentIds: string[] }) {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [documents, setDocuments] = useState<Doc[]>([]);
+  const [error, setError] = useState<Error | null>(null);
+
+  // Fetch data using a more direct approach
+  useEffect(() => {
+    let isMounted = true;
+
+    const fetchDocuments = async () => {
+      try {
+        // Get data from Granola API
+        const url = `https://api.granola.ai/v2/get-documents`;
```

Comment:
Nitpick but backticks used where not needed, double quotes would probably be fine for consistency, but all good either way.